### ISSUE TITLE
Error when optimal scale is zero

### ISSUE THREAD

**Issue by anibalsolon** [neurodata/mgc-matlab#1]
Hi devs,

I am having an issue when running the `MGCFastTest`.
`MGCFastTest(X, y)` where `X` is a matrix 133 x 20 and `y` is a matrix 133 x 1.

I got the error:
```
error: ind2sub: subscript indices must be either positive integers less than 2^31 or logicals
error: called from
    MGCFastTest at line 78 column 9
```

that might be happening since the optimal scale of the relationship is zero (I thought the minimal was 1).

When I run using `MGCPermutationTest`, it works and gives me a p-value of 0.066, and an optimal scale of 16924 (not sure why it is this big since there are only 133 samples).

Thank you!

**Comment by cshen6** [comment]
in matlab, when no signal and optimal scale is 0 (which was just changed
during this weekend), [k,l]=[0,1], so it still runs fine in this case.

My guess is your X or Y is not formatted in double; or something is wrong
for the ns integer -- If you email me your X and Y matrix, I can diagnose
better what went wrong.

Optimal scale being 16924 is fine:
[k,l]=ind2sub([133,133],16924)
yields k=33, l=128, which is all within [1,133] and suggests a nonlinear
relationship since k is small.

**Comment by anibalsolon** [comment]
Cool! Didnt know about ind2sub.

This zip contains X and Y:
[data.zip](https://github.com/neurodata/mgc-matlab/files/2576642/data.zip)

The code to reproduce this behavior is:

```
pkg load statistics
pkg load image
addpath(genpath("../mgc-matlab")) % mgc path
load("cc-omni-parcel.mat"); 
load("y_rank_mgc.mat");

MGCFastTest(data, y(:,1))
```

Thank you for the help @cshen6 !!

**Comment by cshen6** [comment]
Tried on your data, working as intended!

I am using Matlab R2018b on Windows, and the most up-to date github version.

Output a pvalue of 0.023, statistic 0.0559, and optimal scale 16909 =>
[k,l]=[18,128],
quite close to what you have on the permutation test.

So the problem may come from non-code issue, like the matlab version, or
potential mac issue if you are using Mac? (since I only tested on Windows
and Matlab 2017 version or later)
